### PR TITLE
deploy pixiu as dubbo service egress gateway in k8s istio 

### PR DESCRIPTION
**Pull request by ma642**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/config/xds/xds.go [hunk 1]
```diff
@@ -18,6 +18,8 @@
 package xds
 
 import (
+	"github.com/apache/dubbo-go-pixiu/pkg/common/constant"
+	"github.com/mitchellh/mapstructure"
```
Comment (baerwang) [inline pkg/config/xds/xds.go]: import style

File: pkg/config/xds/xds.go [hunk 2]
```diff
@@ -54,20 +56,58 @@ type (
...
+		}
+		return apiclient.CreateEnvoyGrpcApiClient(config, node, a.exitCh, resourceType, apiclient.WithIstioService(dubboServices...))
 	default:
 		logger.Errorf("un-support the api type %s", config.APITypeStr)
 		return nil
 	}
 }
 
+func (a *Xds) readDubboServiceFromListener() ([]string, DiscoverApi, bool) {
+	dubboServices := make([]string, 0)
```
Comment (baerwang) [inline pkg/config/xds/xds.go]:
```go
listeners, err := a.listenerMg.CloneXdsControlListener()
if err != nil {
	logger.Errorf("can not read listener. %v", err)
	return nil, nil, true
}

dubboServices := make([]string, 0)
```

File: pkg/config/xds/xds.go (hunk 1 above)
Comment (mark4z) [reply pkg/config/xds/xds.go]: +1

File: pkg/model/base.go [hunk 3]
```diff
@@ -67,9 +68,10 @@ var (
 	}
 
 	ApiTypeValue = map[string]int32{
-		REST_VALUE:  0,
-		GRPC_VALUE:  1,
-		DUBBO_VALUE: 2,
+		REST_VALUE:      0,
+		GRPC_VALUE:      1,
+		DUBBO_VALUE:     2,
+		IstioGRPC_VALUE: 3,
```
Comment (mark4z) [inline pkg/model/base.go]: ISTIO_GRPC_VALUE?

File: pkg/model/listener.go [hunk 4]
```diff
@@ -28,9 +28,10 @@ const (
 )
 
 const (
-	REST_VALUE  = "REST"
-	GRPC_VALUE  = "GRPC"
-	DUBBO_VALUE = "DUBBO"
+	REST_VALUE      = "REST"
+	GRPC_VALUE      = "GRPC"
+	DUBBO_VALUE     = "DUBBO"
+	IstioGRPC_VALUE = "ISTIO"
```
Comment (mark4z) [inline pkg/model/listener.go]: same with before.

File: pkg/config/xds/apiclient/grpc_envoy.go [hunk 5]
```diff
@@ -0,0 +1,446 @@
+/*
```
Comment (mark4z) [inline pkg/config/xds/apiclient/grpc_envoy.go]: What is the meaning of this file name?

**Review by AlexStocks — APPROVED** [review]

**Comment by AlexStocks** [comment]
pls fix the file confliction.thx.

**Comment by PhilYue** [comment]
is the Sidecar start need special handling?  the relevant code : [sidecar.go](https://github.com/apache/dubbo-go-pixiu/blob/develop/cmd/pixiu/sidecar.go)

File: pkg/config/xds/apiclient/grpc_envoy.go (hunk 5 above)
Comment (ma642) [reply pkg/config/xds/apiclient/grpc_envoy.go]: it's xDS client like envoy over grpc

**Review by mark4z — APPROVED** [review]

**Review by PhilYue — APPROVED** [review]

**Pull request merged**